### ISSUE TITLE
Error with update to Parse 1.14.5

### ISSUE THREAD

**Issue by loxx123** [parse-community/ParseLiveQuery-iOS-OSX#123]
Seeing errors now with Parse Live Query with update to Parse 1.14.5

duplicate interface definition for class 'PFEncoder'
duplicate interface definition for class 'PFPointerOrLocalIdObjectEncoder'
duplicate interface definition for class 'PFNoObjectEncoder'
duplicate interface definition for class 'PFPointerObjectEncoder'
definition of 'PFEncoder' must be imported from module 'Parse.PFEncoder' before it is required
could not build Objective-C module 'ParseLiveQuery'

![Uploading Screen Shot 2017-06-26 at 8.51.25 PM.png…]()

**Comment by flovilmart** [comment]
yes it makes sense, as the encoders are now exposed directly from parse-iOS-SDK. I probably should have released it as a breaking change on the iOS side.

**Comment by loxx123** [comment]
Yep, exactly, I just rolled back to last Parse for now until I have a chance to take a look at reworking code.

**Comment by flovilmart** [comment]
I've unpublished the 1.14.5 version a new 1.15.0 will follow, which is semantically more correct.

**Comment by loxx123** [comment]
Perfect

**Comment by CiraciNicolo** [comment]
Still broken, removing the bridging header seems to fix this because new version of Parse expose that class.

**Comment by flovilmart** [comment]
@CiraciNicolo, can you give more details please?

**Comment by flovilmart** [comment]
Ah right, we need a new version of the LQ SDK so it’s compatible with the latest parse-iOS-SDK now that it’s been released. I’ll craft one ASAP

**Comment by CiraciNicolo** [comment]
Using latest version via Carthage give the errors reported by @larceneaux because in the latest version the Parse iOS SDK export the classes above. Removing the bridging header and the redefinition of the class fix the problem building the project as standalone. I'm still unable to include this library in a complex project.

**Comment by flovilmart** [comment]
Do you set the version on Carthage? Or just use master?

**Comment by CiraciNicolo** [comment]
I'm using the master, `github "parse-community/ParseLiveQuery-iOS-OSX"`.

**Comment by flovilmart** [comment]
that's why, you should probably use the latest tagged version (2.0.0), that's the reason we use semver

**Comment by CiraciNicolo** [comment]
Using the tagged version crashes carthage. My Cartfile is 

```
github "BoltsFramework/Bolts-Swift"
github "parse-community/Parse-SDK-iOS-OSX"
github "parse-community/ParseLiveQuery-iOS-OSX" ~> 2.0.0
<Other package not Parse dependent/dependencies>
```

**Comment by flovilmart** [comment]
Why don’t you pin parse-iOS-SDK? Using master is dangerous

**Comment by CiraciNicolo** [comment]
Using the tagged version makes carthage fail. My Cartfile is 

```
github "BoltsFramework/Bolts-Swift"
github "parse-community/Parse-SDK-iOS-OSX"
github "parse-community/ParseLiveQuery-iOS-OSX" ~> 2.0.0
<Other packages not Parse dependent/dependencies>
```

And the output of `carthage update --platform iOS` is like this:
```
*** Fetching Parse-SDK-iOS-OSX
*** Fetching Bolts-Swift
*** Fetching Bolts-ObjC
*** Fetching Starscream
*** Fetching Parse-SDK-iOS-OSX
*** Checking out Bolts-Swift at "1.3.0"
*** Checking out ParseLiveQuery-iOS-OSX at "2.0.0"
*** Checking out Starscream at "2.0.3"
*** Downloading Parse-SDK-iOS-OSX.framework binary at "1.15.0"
*** Checking out Parse-SDK-iOS-OSX at "1.15.0"
*** Checking out Bolts-ObjC at "1.8.4"
*** Downloading Parse-SDK-iOS-OSX.framework binary at "1.14.2"
*** Checking out Parse-SDK-iOS-OSX at "1.14.2"
A shell task (/usr/bin/env git submodule --quiet update --init --recursive (launched in /Users/<redacted>/Carthage/Checkouts/Parse-SDK-iOS-OSX/Vendor/xctoolchain)) failed with exit code 1:
fatal: Unable to read current working directory: No such file or directory
fatal: /usr/local/Cellar/git/2.13.2/libexec/git-core/git-submodule cannot be used without a working tree.
```

Parse-SDK is referenced 2 times with 2 different versions.

**Comment by flovilmart** [comment]
why don't you force Parse-SDK-iOS-OSX at "1.14.2" ?

**Comment by CiraciNicolo** [comment]
I've already tried, I get the same error as before.

**Comment by CiraciNicolo** [comment]
I was able to build the iphoneos and iphonesimualtor targets from the workspace, lipo them and get a fat binary. Unfortunately seems that some magic is happening with umbrella headers and Xcode cannot find the header of the framework, therefore it's unusable. I can speculate that building it separately leads to a framework with non-mangled class name and the swift runtime cannot locate the classes inside a complex application like mine.

We need a fully compatible version with the latest Parse and Carthage.

**Comment by flovilmart** [comment]
Do you wanna try making a PR on that?

**Comment by CiraciNicolo** [comment]
Yeah, I forked the repo and fixed the problem. I'm still having an issue with Starscream about redefinition of CZLib and CommonCrypto modules. As soon I understand the issue I will PR.

**Comment by flovilmart** [comment]
Awesome! Thanks!

**Comment by CiraciNicolo** [comment]
I can't figure it out, do you have some ideas?

**Comment by CiraciNicolo** [comment]
Okay, I think I got it! My project includes GoogleMaps SDK that already needs libz linked. Removing the Starscream's libz module lead to compile success. I will edit the readme to include this information and then I'll make a PR

**Comment by flovilmart** [comment]
Thanks!

**Comment by Droppix** [comment]
Hi,

I making a PR -> Solves the problem of redefintion (CZLib and CommonCrypto modules).

**Comment by CiraciNicolo** [comment]
Hi, 

redefinition should not be solved with a PR, because it strictly depends of your project: if you use library that already include that modules you have to manually remove the redefinition otherwise it should be left as it is since simple project that depends only on ParseLiveQuery should works out of the box.

**Comment by Droppix** [comment]
Hi,

Many poeple have the same problems with Starscream on others projects, and Starscream have solved this problem with this correction (rename modules).

module SSCZLib [system] {
    header "include.h"
    link "z"
    export *
}
module SSCommonCrypto [system] {
    header "include.h"
    export *
}

In short, just update starscream by switching to version 2.1.1.

**Comment by flovilmart** [comment]
I’ll have a look updating the dependencies, having carharge and Pods don’t make it easy

**Comment by noobs2ninjas** [comment]
Im having the same issue as @Droppix

**Comment by flovilmart** [comment]
I released 2.2.1 with the bumped Starscream version, as recommended by @droppix

**Comment by flovilmart** [comment]
Closing, please reopen if the issue persists.
